konnectivity-client/pkg/client/client.go

```diff
@@ -187,10 +196,16 @@ func (t *grpcTunnel) DialContext(ctx context.Context, protocol, address string)
 	}
 
 	random := rand.Int63() /* #nosec G404 */
+
+	// This channel is closed once we're returning and no longer waiting on resultCh
+	cancelCh := make(chan struct{})
+	defer close(cancelCh)
```

Comment:
I think it would be safer to do the close(cancelCh) under the t.pendingDialLock. I think it would make sense to combine the 2 defers here. 
Eg.
defer func() {
		t.pendingDialLock.Lock()
        close(cancelCh)
		delete(t.pendingDial, random) // Safe to delete missing key.
		t.pendingDialLock.Unlock()
	}()

Reply:
There's no need to lock the close since we're the only closer. I think the close should stay where it is for a couple reasons:
1. an independent `defer` ensures no other panic (even inside the other defer function) keeps us from closing this channel
2. putting `defer close(cancelCh)` immediately after channel creation makes it really easy to see that this channel is always closed on return via all codepaths

Reply:
Sorry should have been more clear. I mean bring the rest of the defer logic here (Line 201), not push this logic down (To line 211). 

Having said that random should be deleted from pendingDial prior to cancelCh being closed which I believe negates my concern.